### PR TITLE
Add Flatpak manifest

### PR DESCRIPTION
**Pull request by t2b3**
This Pull Request adds the option to build a FlatPak and adds some conventions to fulfill the requirements of the Flatpak specification.

In order to successfully build and install it, wxWidgets has to be built and linked statically into TreeSheets but we can benefit from the newly added option to do exactly that.

The stuff related only to Linux is moved into `linux`, i.e. FreeDesktop related stuff.

This can be a step forward to move away from own pre-built binaries that create a lot of issues.

### CODE REVIEW

File: TS/org.aardappel.TreeSheets.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aardappel) [inline TS/org.aardappel.TreeSheets.metainfo.xml:1]: Do all these metadata files have to sit in `TS` ? `TS` is also the distribution folder for other platforms, that clearly don't need this file.

**Comment by aardappel** [comment]
@t2b3 and yes, if you want to maintain it, feel free to make a flatpak available

**Comment by t2b3** [comment]
Hello Wouter
Thanks for your feedback. I removed the submodule and added instructions in the Flatpak manifest to extract the wxWidgets source into the folder `lib/wxWidgets`. In addition to that I moved all the Linux-only stuff out of `TS` and to the folder `linux` and changed the CMakefile to take the changes into account.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "lib/wxWidgets"]
+	path = lib/wxWidgets
```
[the remark this answers is not in the document]
Comment (t2b3) [reply .gitmodules]: Reverted that changed.

File: TS/org.aardappel.TreeSheets.metainfo.xml (hunk 1 above)
Comment (t2b3) [reply TS/org.aardappel.TreeSheets.metainfo.xml]: I put the Linux-only stuff into a separate folder.

**Comment by t2b3** [comment]
Hello Wouter
All checked passed. This PR would be good to go from my side.
I would like to ask you to review it and if it is ok, to merge it. Many thanks in advance!
Best regards
Tobias

**Comment by t2b3** [comment]
Thanks for the merge :-)

**Pull request merged**